plugin-server/src/main/ingestion-queues/anonymous-event-buffer-consumer.ts

```diff
@@ -0,0 +1,105 @@
+import { PluginEvent } from '@posthog/plugin-scaffold'
+import { StatsD } from 'hot-shots'
+import { EachBatchHandler, Kafka } from 'kafkajs'
+import { KafkaProducerWrapper } from 'utils/db/kafka-producer-wrapper'
+
+import { KAFKA_BUFFER, KAFKA_EVENTS_DEAD_LETTER_QUEUE } from '../../config/kafka-topics'
+import { JobName } from '../../types'
+import { status } from '../../utils/status'
+import { GraphileQueue } from '../job-queues/concurrent/graphile-queue'
+import { instrumentEachBatch, setupEventHandlers } from './kafka-queue'
+
+export const startAnonymousEventBufferConsumer = async ({
+    kafka,
+    producer,
+    graphileQueue,
+    statsd,
+}: {
+    kafka: Kafka
+    producer: KafkaProducerWrapper
+    graphileQueue: GraphileQueue
+    statsd?: StatsD
+}) => {
+    /*
+        Consumes from the anonymous event topic, and enqueues the events for
+        processing at a later date as per the message header `processEventAt`.
+
+        We do this delayed processing to allow for the anonymous users that
+        these events are associated to be merged or identified as other
+        "identified" users, at which point to can denormalize data to improve
+        query performance.
+
+        On failure to enqueue we will fail the batch. only on successfully
+        enqueuing an entire batch to Graphile Worker (a PostgreSQL backed async
+        worker library).
+
+        TODO:
+
+        1. distinguish between operational errors and programming errors
+        2. ensure offset only updated if message pushed to DLQ
```

Comment:
is this still valid?

Reply:
Yes at the moment we will update offsets event if the producer hasn't been flushed